riff-raff/app/deployment/actors/DeployGroupRunner.scala

```diff
@@ -47,7 +47,9 @@ class DeployGroupRunner(
   var completed: Set[ValueNode[DeploymentTasks]] = Set.empty
   var failed: Set[ValueNode[DeploymentTasks]] = Set.empty
 
-  def deploymentGraph: Graph[DeploymentTasks] = deployContext.map(_.tasks).getOrElse(Graph.empty[DeploymentTasks])
+  def deploymentGraph: Graph[DeploymentTasks] = failed.foldLeft(deployContext.map(_.tasks).getOrElse(Graph.empty[DeploymentTasks])) {
+    (graph, failedNode) => graph.removeSuccessorValueNodes(failedNode)
+  }
```

Comment:
This change makes the value of deploymentGraph (more) transient. Admittedly it was a `def` but mainly due to the context not being known until the `Start` message was received. Once received the deploymentGraph never changed (so it could actually have been a `var`, set when a `ContextCreated` message is received.

An alternative would be to change `isFinished` to consider the successors of all failed nodes using your `allSuccessors` method. Or update `failed` to include the successors at the time of the call to `markFailed`.